Subscribers list

edwin-pixel-1 opened the pull request:
- Manual merge with master due to conflicts with master branch.
- Solved conflicts with master.
- Integrate Splash activity, MainActivity and SubscribersList Fragment

edwin-pixel-1 committed:
SUBSCRIBERS LIST MVP

- Added MVP basic structure of SubscriberList
- Added Common Classes Utils, Constants, ApiClient and Models

elabcius5 committed:
IMPLEMENTING RECYCLERVIEW, ADAPTER AND FRAGMENT

- Added dependencies library (Picasso, RecyclerView and CardView)
- Implemented adapter for shows data of subscribers list
- Added a RecyclerView for shows login and avatar subscriber
- Added a Fragment that contains a RecyclrView

elabcius5 committed:
IMPLEMENTING RECYCLERVIEW, ADAPTER AND FRAGMENT 2

edwin-pixel-1 committed:
SWIPE AND ADAPTER

- Added swipe refresh to Subscribers list
- Tweaked to the subscriber adapter to use SubscriberItem class
- Fixed package structure of SubscribersList to separate each layout of MVP

edwin-pixel-1 committed:
Merge branch 'master' into SubscribersList

- Manual merging due to conflicts with PullRequest

lucaslabs commented on app/src/main/java/com/github/globant/githubsubscribers/subscriberslist/presenter/SubscribersListPresenterImpl.java:
```diff
@@ -0,0 +1,52 @@
+
+    @Override
+    public void onFailure(String errorMessage) {
+        Utils.debugLog(errorMessage);
```
Show a user-friendly message on the view. Maybe later we can implement a Retry.

lucaslabs commented on app/src/main/java/com/github/globant/githubsubscribers/subscriberslist/view/SubscriberItem.java:
```diff
@@ -0,0 +1,48 @@
+ * @author edwin.cobos
+ * @since 30/08/2016
+ */
+public class SubscriberItem extends RelativeLayout {
```
Why a custom layout?

daniellemariani replied:
I think this class is not needed since you can implement a simple ViewHolder in the adapter

edwin-pixel-1 replied:
I think use this custom layout to separate this from the adapter.
This class is used in the Adapter class.

lucaslabs replied:
+1 for the ViewHolder.

daniellemariani commented on app/src/main/java/com/github/globant/githubsubscribers/commons/utils/ApiClientGithub.java:
```diff
@@ -0,0 +1,47 @@
+    public interface GithubApiInterface {
+
+        /*@GET("repos/googlesamples/android-architecture/subscribers")
+        Call<List<Subscriber>> getSubscribers();*/
```
Commented code should not be commited

lucaslabs commented on app/src/main/java/com/github/globant/githubsubscribers/subscriberslist/view/SubscribersListFragment.java:
```diff
@@ -0,0 +1,91 @@
+        RecyclerView recyclerViewSubscribers = (RecyclerView) viewFragment.findViewById(R.id.recycler_view_subscribers);
+        recyclerViewSubscribers.setLayoutManager(new LinearLayoutManager(getActivity()));
+        recyclerViewSubscribers.setAdapter(subscribersAdapter);
+        loadSubscribers();
```
Do the `loadSubscribers` on the `onResume`.

lucaslabs commented on app/src/main/java/com/github/globant/githubsubscribers/subscriberslist/view/SubscribersListFragment.java:
```diff
@@ -0,0 +1,91 @@
+     * "http://developer.android.com/training/basics/fragments/communicating.html"
+     * >Communicating with Other Fragments</a> for more information.
+     */
+    public interface OnFragmentInteractionListener {
```
Nice!

lucaslabs commented on app/src/main/java/com/github/globant/githubsubscribers/subscriberslist/view/SubscribersListView.java:
```diff
@@ -0,0 +1,19 @@
+    void navigateToUserDetail();
+
+    void showSubscribersList(List<Subscriber> subscriberList);
+
```
Add an extra method to show error.

daniellemariani commented on app/src/main/java/com/github/globant/githubsubscribers/subscriberslist/view/SubscribersAdapter.java:
```diff
@@ -0,0 +1,65 @@
+
+    public class SubscriberViewHolder extends RecyclerView.ViewHolder {
+
+        SubscriberItem item;
```
Define it as a private attribute

lucaslabs commented on app/src/main/java/com/github/globant/githubsubscribers/subscriberslist/view/SubscriberItem.java:
```diff
@@ -0,0 +1,48 @@
+ * @author edwin.cobos
+ * @since 30/08/2016
+ */
+public class SubscriberItem extends RelativeLayout {
```
Also, where are you using this? I'm not seeing any class, code, xml calling this `SubscriberItem`.

edwin-pixel-1 committed:
PR - Fixed Suggestions

- Keep SubscriberItem class

lucaslabs commented on app/src/main/java/com/github/globant/githubsubscribers/subscriberslist/presenter/SubscribersListPresenterImpl.java:
```diff
@@ -0,0 +1,53 @@
+    }
+
+    @Override
+    public void onResume() {
```
IMO Presenter should not have lifecycle method. Please remove them.

edwin-pixel-1 committed:
REMOVED ONRESUME METHOD

- Fixed on onDestroy in SubscriberList Fragment to call onDestroy method in the
  presenter class

edwin-pixel-1 committed:
MODIFIED ADAPTER TO SET ALL IN VIEWHOLDER

- Removed SubscriberItem class
- Modified SubscriberAdapter to set each view in ViewHolder class

lucaslabs commented:
LG

daniellemariani commented on app/src/main/java/com/github/globant/githubsubscribers/commons/models/Subscriber.java:
```diff
@@ -0,0 +1,164 @@
+    @SerializedName("following_url")
+    private final String followingUrl;
+    @SerializedName("gists_url")
+    private final String gistsUrl;
```
Do we need to parse every single attribute from the API for the given requirements? I think gistsUrl as well as starredUrl, eventsUrl, etc. are not needed.

What is the purpose of having them?

daniellemariani commented on app/src/main/java/com/github/globant/githubsubscribers/commons/utils/ApiClientGithub.java:
```diff
@@ -0,0 +1,44 @@
+     */
+    public interface GithubApiInterface {
+
+        @GET("repos/octokit/octokit.rb/subscribers")
```
Based on the requirements. We agreed to work with: Google Android Architecture API (https://api.github.com/repos/googlesamples/android-architecture/subscribers) Check trello card.

daniellemariani commented on app/src/main/java/com/github/globant/githubsubscribers/subscriberslist/view/SubscribersListView.java:
```diff
@@ -0,0 +1,21 @@
+ */
+public interface SubscribersListView {
+
+    void navigateToUserDetail();
```
I think you will need the Subscriber model object as a parameter

edwin-pixel-1 replied:
This will be solved in the next PR, passing just the Username as a parameter.

edwin-pixel-1 committed:
PR - SUBSCRIBER MODEL FIXED

- Removed unused attributes of Subscriber model
- Changed end-point of SubscriberList to: repos/googlesamples/android-architecture/subscribers

juliocatano commented:
LG
